tests/integration/relation_tests/application-charm/src/charm.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+import asyncio
+import logging
+import ssl
+
+import nats
+import ops
+
+logger = logging.getLogger(__name__)
+
+
+class ApplicationCharm(ops.CharmBase):
+    """Application charm that connects to database charms."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # Default charm events.
+        self.framework.observe(self.on.start, self._on_start)
+        self.framework.observe(self.on.client_relation_changed, self._on_client_relation_changed)
+
+    def _on_start(self, _):
+        self.unit.status = ops.ActiveStatus()
+
+    def _on_client_relation_changed(self, event):
```

Comment:
If I understand this correct, the idea is to let the relation fail and the unit set to `error` status to indicate the test has failed?